qutip/random_objects.py

```diff
@@ -1,60 +1,138 @@
-# -*- coding: utf-8 -*-
 """
 This module is a collection of random state and operator generators.
-The sparsity of the ouput Qobj's is controlled by varing the
-`density` parameter.
 """
 
 __all__ = [
     'rand_herm',
-    'rand_unitary', 'rand_unitary_haar',
-    'rand_dm', 'rand_dm_ginibre', 'rand_dm_hs',
+    'rand_unitary',
+    'rand_dm',
     'rand_stochastic',
-    'rand_ket', 'rand_ket_haar',
+    'rand_ket',
     'rand_kraus_map',
-    'rand_super_bcsz', 'rand_super'
+    'rand_super',
+    "rand_super_bcsz",
 ]
 
 import numbers
 
 import numpy as np
+from numpy.random import Generator, SeedSequence, default_rng
```

Comment:
Which version of numpy was the new random interface added in?

Reply:
1.17, not that new.

Reply:
And we'll drop support for numpy 1.16 for v5, so that looks okay.